### ISSUE TITLE
ConnectionWatchdog ignores DNS TTL

### ISSUE THREAD

**Issue by robertvazan** [lettuce-io/lettuce-core#1325]
#### Current Behavior

1. Redis is running on `redis.example.com`, which points to server on IP `111.111.111.111`. Lettuce client is connected to it.
2. During short maintenance window, redis data is moved to server on IP `222.222.222.222` and DNS record for `redis.example.com` is changed to point to it. Traffic from `111.111.111.111` is temporarily proxied to `222.222.222.222`. Lettuce barely notices anything.
3. More than twice the DNS TTL later, server `111.111.111.111` is shut down. Lettuce however keeps trying to reconnect to it, because it caches DNS queries forever.

Client log shows:

```
[lettuce-eventExecutorLoop-1-1] io.lettuce.core.protocol.ConnectionWatchdog - Reconnecting, last destination was redis.example.com/111.111.111.111:6379
```

The only workaround at the moment is to restart the whole client.

#### Expected behavior/code

Lettuce should perform DNS resolution during every reconnect attempt. In the above example, this would result in new connections being made to `222.222.222.222`.

#### Environment

- Lettuce version(s): 5.2.2.RELEASE
- Redis version: 5.0.7

**Comment by mp911de** [comment]
Lettuce uses Java's DNS resolver on each connection attempt so caching originates from the JVM. Please configure your own DNS resolver through ClientResources.

**Comment by robertvazan** [comment]
My app doesn't have SecurityManager installed. I checked that `System.getSecurityManager()` is null. OpenJDK 11 should be defaulting to 30s caching in that case.

The log message indicates that `ConnectionWatchdog` holds resolved `SocketAddress` (host+IP) and no new DNS resolution is attempted during reconnect.

**Comment by mp911de** [comment]
Last destination points to the previously used address, not the new one. Enabling debug logs should show the new target.

**Comment by robertvazan** [comment]
I did some more testing. Java resolver as configured on the client definitely respects DNS TTL. Lettuce also tries to connect to the new IP address, at least according to the simple test I did. All these tests were executed in a thread on the very same client that gave me ConnectionWatchdog logs described above. The reconnect test is simplified compared to full production scenario, but I have hard time imagining what could cause the difference in behavior between the test and production scenario.

I am giving up on this for now. The workaround of restarting clients works. I will add monitoring and if more anomalies pop up, I will reopen the issue.
